src/advanced_inv.cpp

```diff
@@ -1728,60 +1778,69 @@ bool advanced_inventory::query_destination( aim_location &def )
     if( menu.ret >= AIM_SOUTHWEST && menu.ret <= AIM_NORTHEAST ) {
         assert( squares[menu.ret].canputitems() );
         def = static_cast<aim_location>( menu.ret );
+        panes[dest].set_area(squares[def], true);
         uistate.adv_inv_last_popup_dest = menu.ret;
         return true;
     }
     return false;
 }
 
-void advanced_inventory::remove_item( advanced_inv_listitem &sitem )
+void advanced_inventory::remove_item( advanced_inv_listitem &sitem, int count )
 {
     assert( sitem.area != AIM_ALL );        // should be a specific location instead
     assert( sitem.area != AIM_INVENTORY );  // does not work for inventory
-    assert( sitem.area != AIM_WORN );       // yeah what he said
-    assert( sitem.it != nullptr );
-    bool rc = false;
-
-    auto &s = squares[sitem.area];
-    if( s.id == AIM_CONTAINER ) {
-        const auto cont = s.get_container( panes[src].in_vehicle() );
-        assert( cont != nullptr );
-        assert( !cont->contents.empty() );
-        assert( &cont->contents.front() == sitem.it );
-        cont->contents.erase( cont->contents.begin() );
-        rc = true;
-    } else if( sitem.area == AIM_WORN ) {
-        rc = g->u.takeoff( sitem.it );
-    } else if( sitem.from_vehicle ) {
-        rc = s.veh->remove_item( s.vstor, sitem.it );
-    }
-
-    if( !rc ) {
-        // TODO: Make this print an error if it fails
-        g->m.i_rem( s.pos, sitem.it );
+    assert( !sitem.items.empty() );
+    bool rc = true;
+
+    while(count > 0) {
+        auto &s = squares[sitem.area];
+        if( s.id == AIM_CONTAINER ) {
+            const auto cont = s.get_container( panes[src].in_vehicle() );
+            assert( cont != nullptr );
+            assert( !cont->contents.empty() );
+            assert( &cont->contents.front() == sitem.items.front() );
+            cont->contents.erase( cont->contents.begin() );
+        } else if( sitem.area == AIM_WORN ) {
+            rc &= g->u.takeoff( sitem.items.front() );
+        } else if( sitem.from_vehicle ) {
+            rc &= s.veh->remove_item( s.vstor, sitem.items.front() );
+        } else {
+            g->m.i_rem( s.pos, sitem.items.front() );
+        }
+        if(rc == false) {
+            break;
+        }
+        sitem.items.erase(sitem.items.begin());
+        --count;
     }
 }
 
-bool advanced_inventory::add_item( aim_location destarea, item &new_item )
+bool advanced_inventory::add_item( aim_location destarea, item &new_item, int count )
```

Comment:
What happens if this is called with `count` being 10 and only 4 items could be added? How does the caller know that they have to remove 4 items from the source location, but not the originally intended 10 items.

Or if they remove the items first and put back the remaining items, how does the caller know how many items remain.